backend/common.go

```diff
@@ -101,7 +101,7 @@ type DataSourceInstanceSettings struct {
 }
 
 // HTTPClientOptions creates httpclient.Options based on settings.
-func (s *DataSourceInstanceSettings) HTTPClientOptions() (httpclient.Options, error) {
+func (s *DataSourceInstanceSettings) HTTPClientOptions(ctx context.Context) (httpclient.Options, error) {
```

Comment:
since this is also heavily used, can we, at least temporarily, avoid the breaking change? I am suggesting to keep adding changes and when we are finished, make all the breaking changes in the same version (which could be adding the context to multiple functions).

Also, it would make sense to do the same change for AppInstanceSettings?

Reply:
Yeah fair point! Thankfully I think this is the last of them at least from my side.

We could do this, it just means that we'd need to take a more "opt-in" approach by getting people to use the newer `"*WithContext"` version, as opposed to people getting the new approach for free by just fixing their existing signature. In the interest of not slowing ourselves down if we can help it, I'd try avoid IMO. WDYT though @stephaniehingtgen @leandro-deveikis?

>Also, it would make sense to do the same change for AppInstanceSettings?

Good shout! 👍

Reply:
> In the interest of not slowing ourselves down if we can help it, I'd try avoid IMO

I agree with this. Plus this is the kind of change that, in my experience, we intend to keep for a short time but then we forget to update :sweat_smile:  When we actually remove the "without-context" versions,  the result will be the same breaking change (or worst, because the dev will have to find the new func name). 

IMO adding a ctx to the signature is not that heavy,  easy to spot and fix, but I am not the maintainer of the SDK so I don't make the decisions :D I do agree that we should group all the breaking changes together.